nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -288,6 +292,21 @@ Costmap2DROS::on_activate(const rclcpp_lifecycle::State & /*state*/)
       " to become available, tf error: %s",
       robot_base_frame_.c_str(), global_frame_.c_str(), tf_error.c_str());
 
+    // Check timeout
+    if (now() > (start_wait_for_transforms + wait_for_transforms_timeout)) {
+      RCLCPP_ERROR(
+        get_logger(), "Failed to activate %s because transform from %s to %s did not become available before timeout",
+        get_name(), robot_base_frame_.c_str(), global_frame_.c_str());
+      // Deactivate already activated publishers
+      footprint_pub_->on_deactivate();
```

Comment:
Don't deactivate. This will trigger the error state and its not the role of a failed activation to deactivate things according to the ROS 2 lifecycle state machine docs

Reply:
I am deactivating the lifecycle publishers not lifecycle nodes, I don't think this will trigger the error state. According to the lifecycle design doc, a failure in onActivate will result in the inactive state. So I think if I won't deactivate the publishers they will still be active despite the node being in the inactive state. It would be easiest though to just change the order inside the `on_activate` method, so that the transforms are checked at the top of the method.